tests/common/widgets/breadcrumb/options.ts

```diff
@@ -0,0 +1,294 @@
+import {
+  createSearchClient,
+  createMultiSearchResponse,
+  createSingleSearchResponse,
+} from '@instantsearch/mocks';
+import type { BreadcrumbSetup } from '.';
+import type { Act } from '../../common';
+import { screen } from '@testing-library/dom';
+import userEvent from '@testing-library/user-event';
+import { wait } from '@instantsearch/testutils';
+
+const hierarchicalFacets = {
+  'hierarchicalCategories.lvl0': {
+    'Cameras & Camcorders': 1369,
+  },
+  'hierarchicalCategories.lvl1': {
+    'Cameras & Camcorders > Digital Cameras': 170,
+  },
+};
+
+const hierarchicalAttributes = Object.keys(hierarchicalFacets);
+
+const searchClient = createSearchClient({
+  search: jest.fn((requests) => {
+    return Promise.resolve(
+      createMultiSearchResponse(
+        ...requests.map(() =>
+          createSingleSearchResponse({
+            facets: hierarchicalFacets,
+          })
+        )
+      )
+    );
+  }),
+});
+
+/**
+ * Removes rendered DOM discrepancies between flavors when
+ * they don't affect the behavior of the widget
+ * @param element A DOM element to normalize
+ * @returns A normalized DOM element
+ */
+function normalizeSnapshot(element: Element): Element {
+  const normalizedElement = element.cloneNode() as Element;
+  normalizedElement.innerHTML = element.innerHTML
+    // Vue renders a useless extra space between list item elements
+    .replace(/<\/li> <li/g, '</li><li')
+    // Vue renders a useless extra space between last list item element and closing list tag
+    .replace(/<\/li> <\/ul>/g, '</li></ul>');
+
+  return normalizedElement;
+}
```

Comment:
Could this be abstracted into a custom Jest matcher?

Reply:
Oh! I like this idea, I'll try something!

Reply:
I like it being explicit in the test, as we don't want to indiscriminately replace the spacing, right? Otherwise I believe we could also edit the html snapshot serializer to remove explicitly authored spaces in ul/li

Reply:
I kept the definition of the "normalize function" in the common test suite itself, but abstracted the rest in an extension of the `toMatchInlineSnapshot` matcher: https://github.com/algolia/instantsearch/pull/5674/commits/bf6e2b582bb5655e3d3ea5da6b73a7cb752dbaa6.